spec.md

```diff
@@ -692,3 +696,52 @@ Destructuring binding and assignment are not part of ES6, but all binding positi
 ```js
 interface Pattern <: Node { }
 ```
+
+# Concrete syntax
+
+Whitespace, comments, and the exact character sequences comprising Literal nodes are part of the lexical grammar, but not part of the _abstract_ syntax. Their representation is completely optional.
+
+```js
+interface ConcreteNode <: Node {
+    sourceElements: [ ChildReference | Token | Nontoken ];
+}
+```
+
+The `sourceElements` field is a list of input elements and references to child nodes. Any (sub)tree in which `sourceElements` is complete on every node (i.e., there are no unreferenced child nodes and all necessary whitespace and punctuation is included) can be rendered by depth-first concatenation (with respect to references).
+
+## Child-node reference
+
+```js
+interface ChildReference {
+    reference: string;
+}
+```
+
+`reference` names the property identifying a referenced child (e.g., `expression` for ChildReferences on ExpressionStatement nodes), optionally followed by `#next` when identifying the next unreferenced member of a list-valued property (e.g., `body#next` for ChildReferences on BlockStatement nodes).
```

Comment:
This seems like a lot of overhead to identify the node. Why not have a direct reference?